Fix Trying to access array offset on value of type null

DieterHolvoet opened the pull request:
Fixes #32

SenseException reviewed (changes requested):
Please add test(s) that are covering your `null` case.

DieterHolvoet commented:
It's my first time writing a test, so let me know if anything is off.

malarzm commented on tests/Doctrine/Tests_PHP74/Common/Reflection/Dummies/TokenParserNullValueTestClass.php:
```diff
@@ -0,0 +1,13 @@
+
+namespace Doctrine\Tests_PHP74\Common\Reflection\Dummies;
+
+trait TokenParserNullValueTestClass
```
Am I correct assuming this was previously failing due to `static function ($value)` having no name specified? If so then please rename the trait and test to make it obvious what's "wrong"

DieterHolvoet replied:
I changed the name, do you think it's better like this?

malarzm replied:
Way better :+1:

DieterHolvoet commented:
Anything left to do here?

SenseException commented on tests/Doctrine/Tests_PHP74/Common/Reflection/TokenParserAnonymousFunctionTest.php:
```diff
@@ -0,0 +1,24 @@
+        ];
+        $staticReflectionParser = new StaticReflectionParser(TokenParserAnonymousFunctionTestClass::class, new Psr0FindFile($paths));
+
+        $this->assertEquals('', $staticReflectionParser->getDocComment());
```
A small detail:
```suggestion
        self::assertEquals('', $staticReflectionParser->getDocComment());
```

SenseException reviewed (changes requested):
Looks good so far.

I'd like to have another Doctrine reviewer for this PR as a second pair of eyes.

greg0ire commented:
Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/3.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already push. Don't forget the `--force` option otherwise git will try to merge both things together.

DieterHolvoet committed:
Add type check to prevent PHP 7.4 warnings

PHP 7.4 introduces a "Trying to access array offset on value of type ..." warning for accessing null/bool/int/float/resource (everything but array, string and object) as if it were an array. This commit fixes an instance where this warning is emitted.

Fixes #32.

DieterHolvoet commented:
@greg0ire I should have squashed my commits, let me know if I did anything wrong.

greg0ire commented:
It's better 👍 On to the next issue: the commit message itself.

Please improve your commit message according to [the contributing guide](https://www.doctrine-project.org/contribute/index.html#working-on-topic-branches).

DieterHolvoet commented:
Changed the commit message.

DieterHolvoet commented:
Any update on this?

malarzm commented:
@DieterHolvoet sorry for long time wait! I've changed the base branch to `1.1.x` since that's the branch you've found the error on and it's still maintained, but now PR has additional commits. Could you please rebase your work onto `1.1.x`? Please ping me once that's done and I'll merge it right away :)

DieterHolvoet commented:
@malarzm Should be done now

greg0ire commented:
There are commits that do not belong to you in here. I'm going to rebase your branch.

greg0ire commented:
Ah you didn't allow edit from maintainers, so I cannot. Can you please use `git reset --hard origin/1.1.x && git cherry-pick d2ca550 && git push -f`?

DieterHolvoet commented:
I did allow edits from maintainers, but I'll try. I wonder what I did wrong...

greg0ire commented:
Thanks @DieterHolvoet !
